### PR TITLE
Update: Enabled scheduling in directory view with delivery-rule consumer

### PR DESCRIPTION
**Pull request by mattgraham1995**
Enables the schedule button in the directory list view and adds a consumer for delivery-rule actions.

### CODE REVIEW

File: packages/reports/addon/components/common-actions/schedule.js [hunk 1]
```diff
@@ -142,6 +153,29 @@ export default Ember.Component.extend({
       }
     },
 
+    onDelete() {
```
Comment (balajimanoharan) [inline packages/reports/addon/components/common-actions/schedule.js]: jsdoc pls

File: packages/reports/addon/components/common-actions/schedule.js [hunk 2]
```diff
@@ -142,6 +153,29 @@ export default Ember.Component.extend({
...
+      let deliveryRule = get(this, 'localDeliveryRule'),
+        deletePromise = new RSVP.Promise((resolve, reject) => {
+          this.attrs.onDelete(deliveryRule, { resolve, reject });
+        });
+
+      return deletePromise
+        .then(() => {
+          get(this, 'naviNotifications').add({
```
Comment (balajimanoharan) [inline packages/reports/addon/components/common-actions/schedule.js]: Can we add a comment to say `Add page notification`, just to differentiate between this and the `set('notification')`?

File: packages/reports/addon/consumers/delivery-rule.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import ActionConsumer from 'navi-core/consumers/action-consumer';
+import { DeliveryRuleActions } from 'navi-reports/services/delivery-rule-action-dispatcher';
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default ActionConsumer.extend({
+  /**
+   * @property {Service} naviNotifications
+   */
+  naviNotifications: service(),
```
Comment (balajimanoharan) [inline packages/reports/addon/consumers/delivery-rule.js]: Do we need this?

File: packages/reports/addon/components/common-actions/schedule.js [hunk 4]
```diff
@@ -118,8 +126,11 @@ export default Ember.Component.extend({
         });
 
         set(this, 'attemptedSave', false);
-        this.attrs
-          .onSave(deliveryRule)
+        let savePromise = new RSVP.Promise((resolve, reject) => {
```
Comment (balajimanoharan) [inline packages/reports/addon/components/common-actions/schedule.js]: Why do we need the promise wrapper?

File: packages/reports/addon/helpers/delivery-rule-action.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+   * @param {Array} array with the name of the action to dispatch
+   * @returns {Function} Closure action
+   */
+  compute([action, ...params]) {
+    let actionName = DeliveryRuleActions[action];
+    assert(`The action name "${action}" is not a valid delivery rule action`, actionName);
+    let deliveryRuleAction = ((...parameters) => join(() => get(this, 'deliveryRuleActionDispatcher').dispatch(actionName, ...params, ...parameters))),
+        actionId = CLOSURE_ACTION_MODULE.ACTION;
+
+    //Make the report action a "real" action
```
Comment (balajimanoharan) [inline packages/reports/addon/helpers/delivery-rule-action.js]: Need a comment update

File: packages/reports/addon/routes/reports.js [hunk 6]
```diff
@@ -1,100 +1,6 @@
...
-            message: `OOPS! An error occurred while deleting report "${reportName}"`,
-            type: 'danger',
-            timeout: 'short'
-          });
-        });
-    }
-  }
-});
+import Route from '@ember/routing/route';
+export default Route;
```
Comment (balajimanoharan) [inline packages/reports/addon/routes/reports.js]: Can we do `Route.extend()`?

File: packages/reports/addon/components/common-actions/schedule.js (hunk 4 above)
Comment (mattgraham1995) [reply packages/reports/addon/components/common-actions/schedule.js]: The save is being handled through a consumer now. This promise will resolve or reject once the save succeeds or fails, then we can push the appropriate notification from this component.

**Pull request merged**